platform/platform.h

```diff
@@ -55,31 +55,37 @@ typedef _interval_t interval_t;
  */
 typedef _microstep_t microstep_t;
 
-// Define mutex functions even though the runtime is unthreaded because they are
-// needed to support interrupts.
-
 /**
- * Initialize a mutex.
- * 
+ * Enter a critical section where logical time and the event queue are guaranteed
+ * to not change unless they are changed within the critical section.
+ * In platforms with threading support, this normally will be implemented
+ * by acquiring a mutex lock. In platforms without threading support,
+ * this can be implemented by disabling interrupts.
+ * Users of this function must ensure that lf_init_critical_sections() is
+ * called first and that lf_critical_section_exit() is called later.
  * @return 0 on success, platform-specific error number otherwise.
  */
-extern int lf_mutex_init(lf_mutex_t* mutex);
+extern int lf_critical_section_enter();
 
 /**
- * Lock a mutex.
- * 
+ * Exit the critical section entered with lf_lock_time().
  * @return 0 on success, platform-specific error number otherwise.
  */
-extern int lf_mutex_lock(lf_mutex_t* mutex);
+extern int lf_critical_section_exit();
 
-/** 
- * Unlock a mutex.
- * 
+/**
+ * Notify any listeners that an event has been created.
+ * The caller should call lf_critical_section_enter() before calling this function.
  * @return 0 on success, platform-specific error number otherwise.
  */
-extern int lf_mutex_unlock(lf_mutex_t* mutex);
-
+extern int lf_notify_of_event();
 
+/**
+ * Acknowledge any event notifications that may have occurred.
```

Comment:
"Acknowledge" usually means that something or someone is notified.  I think this might be misnamed.
```suggestion
 * Clear the flag indicating that events have occurred.
```

Reply:
Also, this does not seem to be called anywhere.